controller/konnect/reconciler_generic_type_specific.go

```diff
@@ -47,11 +57,34 @@ func handleTypeSpecific[
 	return isProblem, res, err
 }
 
+func handleKonnectGatewayControlPlaneSpecific(
+	ctx context.Context,
+	sdk sdkops.SDKWrapper,
+	kgcp *konnectv1alpha1.KonnectGatewayControlPlane,
+) (updated bool, err error) {
+	// If it's not set it means that first time we are reconciling the KonnectGatewayControlPlane,
+	// in subsequent reconciliations it should be set. Otherwise it will be reported and everything
+	// in this helper is irrelevant.
+	kgcpID := kgcp.GetKonnectID()
+	if kgcpID == "" {
+		return updated, nil
```

Comment:
Here we're using a name return variable but below we're using verbatim bool values. Can we unify this? (I'd suggest to use the latter).